libbeat/common/docker/watcher.go

```diff
@@ -316,7 +338,8 @@ func (w *watcher) listContainers(options types.ContainerListOptions) ([]*Contain
 		// If there are no network interfaces, assume that the container is on host network
 		// Inspect the container directly and use the hostname as the IP address in order
 		if len(ipaddresses) == 0 {
-			info, err := w.client.ContainerInspect(w.ctx, c.ID)
+			logp.Debug("docker", "Inspect container %s", c.ID)
+			info, err := w.client.ContainerInspect(ctx, c.ID)
 			if err == nil {
 				ipaddresses = append(ipaddresses, info.Config.Hostname)
 			} else {
```

Comment:
In here problems are only logged, and rest of process is going on - possibly missing some important data. I think error should be thrown to higher level and retry fetching complete data.

I attach fragment of debug log when dockerd not responds for single container 81c1bf9d22ac4133d032d3f4b7b69965cc7b6815c34f817c38216afc09fcff0d:
```
2019-05-29T10:23:37.763+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container e302f08e88e9ec6c0c33f86a8c7e8e19e3432ba83e2426a694cba68c9ff88755
2019-05-29T10:23:37.764+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container ffb0b39ebbcac8942ca25b583333915c8f130e4cb97d07689dc0c81ddda0af8a
2019-05-29T10:23:37.764+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 1bbd6cfbf6bbceeb44581abec8279474a4885d77fb365c820ae192bda9832287
2019-05-29T10:23:37.765+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container e84b4c90b495006e8f364c464bed289cb929a24c3026cf01fa8c3d6700f2aa14
2019-05-29T10:23:37.766+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 81c1bf9d22ac4133d032d3f4b7b69965cc7b6815c34f817c38216afc09fcff0d
2019-05-29T10:23:57.685+0200    WARN    docker/watcher.go:346   unable to inspect container 81c1bf9d22ac4133d032d3f4b7b69965cc7b6815c34f817c38216afc09fcff0d due to error context deadline exceeded
2019-05-29T10:23:57.686+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 298933efede203f12cc9c54c7f84d3738508beac56a0f87a48642671684de50d
2019-05-29T10:23:57.686+0200    WARN    docker/watcher.go:346   unable to inspect container 298933efede203f12cc9c54c7f84d3738508beac56a0f87a48642671684de50d due to error context deadline exceeded
2019-05-29T10:23:57.686+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 8bbb2fafed0b5ad6c89180264082ffefe2851766db4869ffea3b5f02a1d32b71
2019-05-29T10:23:57.686+0200    WARN    docker/watcher.go:346   unable to inspect container 8bbb2fafed0b5ad6c89180264082ffefe2851766db4869ffea3b5f02a1d32b71 due to error context deadline exceeded
2019-05-29T10:23:57.686+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container a27d8e49feb9da282efd37621abeb4868875fa8f5600e92580ccfd7158f58752
2019-05-29T10:23:57.686+0200    WARN    docker/watcher.go:346   unable to inspect container a27d8e49feb9da282efd37621abeb4868875fa8f5600e92580ccfd7158f58752 due to error context deadline exceeded
2019-05-29T10:23:57.686+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container f375017813bee1b80b2346ff6756839bac76e05fd6c63951d3e11a01a77919c4
2019-05-29T10:23:57.686+0200    WARN    docker/watcher.go:346   unable to inspect container f375017813bee1b80b2346ff6756839bac76e05fd6c63951d3e11a01a77919c4 due to error context deadline exceeded
2019-05-29T10:23:57.686+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 5789d514c615404b79322d22a2df939e356b161d1311d545c8cb448737ee2fa4
2019-05-29T10:23:57.686+0200    WARN    docker/watcher.go:346   unable to inspect container 5789d514c615404b79322d22a2df939e356b161d1311d545c8cb448737ee2fa4 due to error context deadline exceeded
2019-05-29T10:23:57.687+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 4e5ac2581157f675b5f1f53d889999cc2289a02f56810e926ed34c5b2a59be31
2019-05-29T10:23:57.687+0200    WARN    docker/watcher.go:346   unable to inspect container 4e5ac2581157f675b5f1f53d889999cc2289a02f56810e926ed34c5b2a59be31 due to error context deadline exceeded
2019-05-29T10:23:57.687+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container d9f75d8212e096f39803969b86844eac2deefc35c9c431829235929dc5915d70
2019-05-29T10:23:57.687+0200    WARN    docker/watcher.go:346   unable to inspect container d9f75d8212e096f39803969b86844eac2deefc35c9c431829235929dc5915d70 due to error context deadline exceeded
2019-05-29T10:23:57.687+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container c22f089a00417eb179bcfbf910e46099bdabbe21dceb4b4eaf12ea0dc7bb2fc7
2019-05-29T10:23:57.687+0200    WARN    docker/watcher.go:346   unable to inspect container c22f089a00417eb179bcfbf910e46099bdabbe21dceb4b4eaf12ea0dc7bb2fc7 due to error context deadline exceeded
2019-05-29T10:23:57.687+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 13b585292c652a2ad86c6d4f027367f54f8ec286a2404336b33a3592f69a8bfc
2019-05-29T10:23:57.687+0200    WARN    docker/watcher.go:346   unable to inspect container 13b585292c652a2ad86c6d4f027367f54f8ec286a2404336b33a3592f69a8bfc due to error context deadline exceeded
2019-05-29T10:23:57.687+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container cfc58cdce29c27887a2d7de1664e0221c827e0c60e618f9002e5ede369439189
2019-05-29T10:23:57.687+0200    WARN    docker/watcher.go:346   unable to inspect container cfc58cdce29c27887a2d7de1664e0221c827e0c60e618f9002e5ede369439189 due to error context deadline exceeded
2019-05-29T10:23:57.687+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 2f6cc51fe298b71252c3bd6132a3539f18ffa9f40c11b0d85cc2394cfe6a7bac
2019-05-29T10:23:57.687+0200    WARN    docker/watcher.go:346   unable to inspect container 2f6cc51fe298b71252c3bd6132a3539f18ffa9f40c11b0d85cc2394cfe6a7bac due to error context deadline exceeded
2019-05-29T10:23:57.687+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 42d5fa4e1f98e207180eda0d8e82c9c942d92eb3d1267958aee697f05ac42a67
2019-05-29T10:23:57.687+0200    WARN    docker/watcher.go:346   unable to inspect container 42d5fa4e1f98e207180eda0d8e82c9c942d92eb3d1267958aee697f05ac42a67 due to error context deadline exceeded
2019-05-29T10:23:57.687+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 97ba5ecb0319995fa2742449380ee5bbd13d5b5454b8e75e10ebc371d6f40479
2019-05-29T10:23:57.688+0200    WARN    docker/watcher.go:346   unable to inspect container 97ba5ecb0319995fa2742449380ee5bbd13d5b5454b8e75e10ebc371d6f40479 due to error context deadline exceeded
2019-05-29T10:23:57.688+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container f673cf45c2b07a03d817ef69ada6e3f208fc7a5000cde8af6e2034d52f38ece3
2019-05-29T10:23:57.688+0200    WARN    docker/watcher.go:346   unable to inspect container f673cf45c2b07a03d817ef69ada6e3f208fc7a5000cde8af6e2034d52f38ece3 due to error context deadline exceeded
2019-05-29T10:23:57.688+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 7814278e7e81852047b1810fc6a6f3ca5c77fa33168ae1fd7a2dd5bd08b8247b
2019-05-29T10:23:57.688+0200    WARN    docker/watcher.go:346   unable to inspect container 7814278e7e81852047b1810fc6a6f3ca5c77fa33168ae1fd7a2dd5bd08b8247b due to error context deadline exceeded
2019-05-29T10:23:57.688+0200    DEBUG   [docker]        docker/watcher.go:341   Inspect container 9f187b5c30cb2ceb43907a2513e093f1039be42c230c33d70a601e9c3173dc5d
2019-05-29T10:23:57.688+0200    WARN    docker/watcher.go:346   unable to inspect container 9f187b5c30cb2ceb43907a2513e093f1039be42c230c33d70a601e9c3173dc5d due to error context deadline exceeded

```

Reply:
Actually `inspect` query is only used to attach container IP addresses, rest of data is ok, so maybe it is not that big problem. with separate ctx for every inspect call current behavior seems acceptable